src/Store.js

```diff
@@ -0,0 +1,26 @@
+import { AsyncStorage } from 'react-native';
+import { persistStore, persistReducer } from 'redux-persist';
+import { applyMiddleware, createStore } from 'redux';
+import thunk from 'redux-thunk';
+import { createReactNavigationReduxMiddleware } from 'react-navigation-redux-helpers';
+import reducers from './reducers';
+
+// Create middleware and connect
+const appNavigatorMiddleware = createReactNavigationReduxMiddleware(state => {
+  return state.nav;
+}, 'root');
+
+const persistConfig = {
+  keyPrefix: '',
+  key: 'root',
+  storage: AsyncStorage,
+  blacklist: ['nav'],
+};
+
+const persistedReducer = persistReducer(persistConfig, reducers);
+
+const store = createStore(persistedReducer, {}, applyMiddleware(thunk, appNavigatorMiddleware));
+
+const persistedStore = persistStore(store);
+
+export { store, persistedStore };
```

Comment:
I like this refactor of the store.  I wonder if the file structure could use an improvement though? Personally I like having all of the redux related files consolidated in a 'store' or 'redux' folder. Currently it seems just so spread, makes things a little confusing to understand. What do you think?

Reply:
I agree.

Reply:
store.js and reducers.js are composition files that would put all the other files together. That's why they go to the root of src folder. They can be like a table of content.

I am not fan of creating redux folder. Store folder also does not sound good to me. We can create actions and reducers folders and keep all the actions and reducers there. That would be the standard redux way to do it.

Reply:
Resolved with dfa7207